src/Map.ts

```diff
@@ -752,6 +762,37 @@ export class Map extends maplibregl.Map {
     });
 
     this.telemetry = new Telemetry(this);
+
+    // Set default value for defaultControls
+    options = {
+      defaultControls: true,
+      ...options,
+    };
+
+    // Check for potential control configuration issues
+    this.checkControlsConfiguration(options);
+  }
+
+  private checkControlsConfiguration(options: MapOptions): void {
+    if (options.defaultControls === false) {
+      const controlOptions = [
+        { name: "navigationControl", value: options.navigationControl },
+        { name: "geolocateControl", value: options.geolocateControl },
+        { name: "terrainControl", value: options.terrainControl },
+        { name: "projectionControl", value: options.projectionControl },
+        { name: "scaleControl", value: options.scaleControl },
+        { name: "fullscreenControl", value: options.fullscreenControl },
+      ];
+
+      controlOptions.forEach(({ name, value }) => {
+        if (value !== undefined && value !== false) {
+          console.warn(
+            `Warning: ${name} is set but will be ignored because defaultControls is false. ` +
+              `If you want to use ${name}, either remove defaultControls: false or add the control manually using addControl().`,
+          );
```

Comment:
This is probably out of scope of this PR and I know it was part of the codebase before and is not your changes, but the conflating of falsey values (here and on L618, L628 etc) kind of gives me the heebie jeebies. Feels like a bit of code smell. Although I don't know what the alternative would be given the control defaults. 🤔

Reply:
Good shout on adding the console warning tho 👍